components/net/fetch/methods.rs

```diff
@@ -732,9 +735,30 @@ async fn scheme_fetch(
                 if let Ok(file) = File::open(file_path.clone()) {
                     if let Ok(metadata) = file.metadata() {
                         if metadata.is_dir() {
-                            return Response::network_error(NetworkError::Internal(
-                                "Opening a directory is not supported".into(),
-                            ));
+                            // Skip past the "file://" prefix, otherwise from_directory_path
+                            // will return an error.
+                            let url_path = &url.as_str()[7..];
```

Comment:
I've reworked (in the fifth revision in this pull request) this section of code so that it no longer treats the entire URL as a string, only the `path()` as you suggested.